lib/engine/step/special_token.rb

```diff
@@ -60,11 +60,12 @@ def process_place_token(action)
             true
           end
 
+        connected = special_ability.respond_to?(:reachable) ? special_ability.reachable : false
```

Comment:
Would this work instead? I think we want to avoid adding new uses of `respond_to?` where possible.

```suggestion
        connected = special_ability.type == :token ? special_ability.reachable : false
```

Reply:
Depends on how generic we want that to be I guess. I suspect `special_token` could be used by more abilities in the future. But I can see if `.type == :token` works, I wasn't keen on `respond_to?` either.